plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchTableHandle.java

```diff
@@ -39,6 +43,10 @@
     private final TupleDomain<ColumnHandle> constraint;
     private final Optional<String> query;
     private final OptionalLong limit;
+    // for group by fields
+    private final List<TermAggregation> termAggregations;
+    // for aggregation methods and fields
+    private final List<MetricAggregation> metricAggregations;
```

Comment:
The semantics of aggregations vs constraint/limit/query need to be clarified. Do the aggregations apply before of after the limit? Before or after the constraint? How do they relate to `query`?

Reply:
@martint The `query` and the rest are mutually exclusive. When `query` does not exist, `constraint` must occur before aggregation, but whether limit occurs before or after aggregation. This point really needs to be clarified. Can you please provide some suggestions?